pkg/server/server.go

```diff
@@ -272,3 +290,47 @@ func (s *Server) newEndpointsServer(catalog catalog.Catalog, svidRotator svid.Ro
 func (s *Server) caCertsPath() string {
 	return path.Join(s.config.DataDir, "certs.json")
 }
+
+func (s *Server) validateTrustDomain(ctx context.Context, ds datastore.DataStore) error {
+	trustDomain := s.config.TrustDomain.Host
+
+	fetchResponse, err := ds.ListRegistrationEntries(ctx, &datastore.ListRegistrationEntriesRequest{})
```

Comment:
I think this will be problematic for production deployments, since we will be scanning the whole table every time we start/restart... we should add a pagination option or something (perhaps as part of a separate PR/Issue?)